includes/hook_handlers/order-hook-handler.php

```
@@ -67,6 +74,50 @@ function checkout_update_order_hook($posted_data)
     }
 }
 
+/**
+ * Add a custom action to order actions select box on edit order page
+ * Only added for Simpl orders
+ *
+ * @param array $actions order actions array to display
+ * @return array - updated actions
+ */
+function simpl_add_sync_order_action( $actions ) {
+	global $theorder;
+
+    // Render the option for simpl orders only
+    if('yes' == get_post_meta( $theorder->id, SIMPL_ORDER_METADATA, true ) ) {
+        $actions['wc_custom_order_action'] = SYNC_ORDER_ACTION_TEXT; // Remove hardcoding - create a constant    
+    }
+    return $actions;
+}
+
+function simpl_process_sync_order_action( $order ) {
+    if ($order->meta_exists('simpl_order_id')) {
+        $order_data = fetch_order_data($order);
+
+        $request["topic"] = "order.updated";
+        $request["resource"] = "order";
+        $request["event"] = "updated";
+        $request["data"] = $order_data;
+
+        $checkout_3pp_client = new SimplCheckout3ppClient();
+        try {
+            $simplHttpResponse = $checkout_3pp_client->post_hook_request($request);
+        } catch (\Throwable $th) { 
+            error_log(print_r($th, TRUE));
+        }
+
+        // add the sync timestamp to order note
+        if (simpl_is_success_response($simplHttpResponse)) {
+            $message = sprintf( 'Order synced with Simpl Checkout', wp_get_current_user()->display_name );
+            $order->add_order_note( $message );
+        } else {
+            $message = sprintf( 'Order sync with Simpl Checkout failed!', wp_get_current_user()->display_name );
+            $order->add_order_note( $message );
```

Lets throw an exception in case of failure. Order note may get unnoticed.

---

throwing an exceptions renders the fatal error page since in this hook , exceptions aren't handled so it becomes and unhandled exception and crashes the page

---

aah, got it.